### PR TITLE
switch logging to tracing ecosystem

### PR DESCRIPTION
**Pull request by flokli**
This switches:
 - from android_logger to tracing-android
 - from oslog to tracing-oslog
 - from log to the default tracing log infrastructure

Most of the noise in here is changing the imports from `log::{…}` to `tracing::{…}`.

In terms of log setup, for now this tries to stick with the existing behaviour:

 - myceliumd now uses `EnvFilter` and the `compact()` default formatter, instead of `prety_env_logger`. Its CLI args are kept as-is. `RUST_LOG` env var still works, too.
 - The `setup_the_logger` function in the `mobile` crate is renamed to `setup_logging`.

I didn't change any of the actual log messages yet.
In a future PR, I'd like to make more use of the `#[instrument]` macro to annoate some more fields, as well as enable OTLP (behind a feature flag).

Note I did not check building for IOS and Android for the `mobile` crate.

### CODE REVIEW

**Comment by iwanbk** [comment]
> Note I did not check building for IOS and Android for the mobile crate.

Oh, i can help to add github actions CI for this next week.

For now, you can do some quick check for iOS
```bash
cd mobile
rustup target add aarch64-apple-ios
cargo build --target aarch64-apple-ios
```

for android, you need to install `cargo ndk` first
```bash
cargo install cargo-ndk
```
but for that work, you also need to install Android NDK first:
- install Android Studio
- install NDK from android Studio

If the `ndk` things seems very long, i think for now you can simply do the iOS work and i can help to check with the Android NDK

**Comment by iwanbk** [comment]
> switch logging to tracing ecosystem

Also, why we need to do this?

**Comment by flokli** [comment]
> > switch logging to tracing ecosystem
> 
> Also, why we need to do this?

Mostly for everything the tracing crate also provides, like all the span functionality (span hierarchies/fields attached elsewhere to the current span), as well as the large ecosystem of other tracing subscribers, like OTLP/Tracy.

I'm happy to take another pass through the code base making more use of this functionality in a followup PR, but wanted to keep the diff minimal.

Thanks for the info on how to test the mobile crate. I'm currently back to a bad connection due to travels, but can check once that's better again. I'd appreciate it being wired into CI though.

**Comment by iwanbk** [comment]
> I'd appreciate it being wired into CI though.

added at #256

**Comment by flokli** [comment]
Thanks! I cherry-picked it into here, and got all of CI to [pass](https://github.com/threefoldtech/mycelium/actions/runs/9156709816?pr=254). Happy to rebase this another time one #256 is merged.

**Comment by LeeSmet** [comment]
I'll merge this once it's rebased on master (now that #256 is merged). One thing I noticed is that there is no longer a directive to filter out non-mycelium logs. When running this PR code, no such logs were printed though. Does the tracing crate handle this by default, or is it just a lack of tracing support in dependecies?

**Comment by flokli** [comment]
> I'll merge this once it's rebased on master (now that #256 is merged).

Thanks! Will do the rebase in some hours, once I'm back on the computer.

One thing I noticed is that there is no longer a directive to filter out non-mycelium logs. When running this PR code, no such logs were printed though. Does the tracing crate handle this by default, or is it just a lack of tracing support in dependecies?

We currently set a global log level [here](https://github.com/threefoldtech/mycelium/pull/254/commits/39bbf96a3b9ee9a2e7955a0ec391d65c77d81b50#diff-12582e67fe590f5485e453546fd9c83e2460b4c8661cf875749bb12eb6f07de0R229), and still support more granular filtering via `RUST_LOG` env var (through [EnvFilter](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.EnvFilter.html#directives)). This should be easy to tweak.

**Comment by flokli** [comment]
> When running this PR code, no such logs were printed though. Does the tracing crate handle this by default, or is it just a lack of tracing support in dependecies?

Ah, I think now I understand your question properly :-) If we have crate dependencies only using log, not tracing, and want its logs to also be collected, we'd need to use https://docs.rs/tracing-log/latest/tracing_log/#convert-log-records-to-tracing-events.

Do you have an example you noticed this with, a log line that's now missing? Happy to add these bits to, either to this PR or a followup.

**Review by LeeSmet — APPROVED** [review]
lgtm

**Comment by LeeSmet** [comment]
After merging, I found out that it crashes on startup because a global logger is being set twice. It turns out that tracing has a feature gate, enabled by default, which basically has the exact behavior to install a tracing_log logger. I've reverted the second commit of this PR, and it seems that logs from other crates are indeed properly processed (though there aren't a lot) :sweat_smile:

**Comment by flokli** [comment]
Ok! Thanks for fixing this up!

**Pull request merged**
